plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFusionEnv.groovy

```diff
@@ -0,0 +1,257 @@
+package io.seqera.tower.plugin
+
+import com.google.gson.Gson
+import dev.failsafe.Failsafe
+import dev.failsafe.RetryPolicy
+import dev.failsafe.event.EventListener
+import dev.failsafe.event.ExecutionAttemptedEvent
+import dev.failsafe.function.CheckedSupplier
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.seqera.tower.plugin.exception.BadResponseException
+import io.seqera.tower.plugin.exception.UnauthorizedException
+import io.seqera.tower.plugin.exchange.LicenseTokenRequest
+import io.seqera.tower.plugin.exchange.LicenseTokenResponse
+import nextflow.Global
+import nextflow.Session
+import nextflow.SysEnv
+import nextflow.exception.AbortOperationException
+import nextflow.fusion.FusionConfig
+import nextflow.fusion.FusionEnv
+import nextflow.platform.PlatformHelper
+import nextflow.util.Threads
+import org.pf4j.Extension
+
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.Executors
+import java.util.function.Predicate
+
+/**
+ * Environment provider for Platform-specific environment variables.
+ *
+ * @author Alberto Miranda <alberto.miranda@seqera.io>
+ */
+@Slf4j
+@Extension
+@CompileStatic
+class TowerFusionEnv implements FusionEnv {
+
+    // The path relative to the Platform endpoint where license-scoped JWT tokens are obtained
+    private static final String LICENSE_TOKEN_PATH = 'license/token/'
+
+    // Server errors that should trigger a retry
+    private static final List<Integer> SERVER_ERRORS = [429, 500, 502, 503, 504]
```

Comment:
I'm not sure how I feel about retrying 503 and 504 errors, if Seqera Platform is under heavy load wouldn't this make it worse?

Reply:
I took this from [Wave's client](https://github.com/nextflow-io/nextflow/blob/ee252173bf823d4bbef5522bd726ccd4df595a7d/plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveClient.groovy#L788-L795) to follow the existing approach. I don't have a specific preference :man_shrugging: (cc @pditommaso)

Reply:
That's why it's used an exponential backoff. Actually think it should be added 408 (connection timeout) to that list

Reply:
Makes sense, consider this resolved :+1:

Reply:
Error 408 added with 6ce5be3dc2a152d2642aba4af779bf3254d8b2c4